engine/collection/compliance/core.go

```diff
@@ -330,8 +334,7 @@ func (c *Core) processBlockProposal(proposal *cluster.Block) error {
 	if err != nil {
 		if state.IsInvalidExtensionError(err) {
```

Comment:
I don't have a hard preference but let me give you reasoning why I do this way, if it's not convincing enough I am happy to change. 
1. Writing it with `if-else if` implies that conditions are mutually exclusive, and only one branch should be executed from a set. It more explicit I would say.
2. It has performance benefit(not in this case though) since when one condition from all `if-else if` statement is satisfied then all other branches won't be calculated, with separate `if` statements you will need to make each check even if the first one has already succeeded.

So unless we can match multiple branches(which isn't the case in this situation) I prefer to use `if-else if`.

Reply:
your code, your stylistic decisions -- happy with the code how it is